src/Controller/GameController.php

```diff
@@ -3,26 +3,135 @@
+        $this->translator = $translator;
     }
 
-    public function random(): Response
-    {        
-        $game = $this->gameManipulator->createGame();
+    /**
+     * @ParamConverter("game", options={"mapping": {"hash": "hash"}})
+     */
+    public function index(Request $request, Game $game): Response
+    {
+        if ($game->isAbandoned()) {
+            return $this->render('abandoned.html.twig');
+        }
+
+        $abandonForm = $this->createForm(AbandonGameType::class);
+        $abandonForm->handleRequest($request);
+
+        if ($abandonForm->isSubmitted() && $abandonForm->isValid()) {
+            $this->gameManipulator->abandonGame($game);
+
+            return $this->redirectToRoute('index');
+        }
+
+        $triggerViews = $this->createTriggersForms($game);
+
+        $trigger = $this->createForm(ShootShipType::class)->handleRequest($request);
+
+        if ($trigger->isSubmitted() && $trigger->isValid()) {
+
+            $this->manageShoot($trigger->getData(), $game);
+
+            return $this->redirectToRoute('game_index', ['hash' => $game->getHash()]);
+        }
+
+        $hits = $this->gameManipulator->getCurrentPlayerHits($game);
+
+        $ships = $this
+            ->getDoctrine()
+            ->getRepository(Ship::class)
+            ->getCurrentPlayerShips($game);
+
+        $shoots = $this
+            ->getDoctrine()
+            ->getRepository(Shoot::class)
+            ->getCurrentPlayerShoots($game);
+            
+        $opponentShips = $this
+            ->getDoctrine()
+            ->getRepository(Ship::class)
+            ->getPlayerShips($game, $this->gameManipulator->getOpponentPlayer($game));
 
         return $this->render('game.html.twig', [
             'game' => $game,
+            'ships' => $ships,
+            'opponent_ships' => $opponentShips,
+            'opponent_ships_sunk' => $this->gameManipulator->getOpponentShipsSunk($game),
+            'hits' => $hits,
+            'shoots' => $shoots,
+            'triggers' => $triggerViews,
+            'abandon_form' => $abandonForm->createView()
         ]);
     }
+
+    /**
+     * Creates all the form views for shooting the opponent
+     * 
+     * @param Game $game
+     * 
+     * @return array
+     */
+    private function createTriggersForms(Game $game)
+    {
+        $triggerViews = [];
+
+        foreach (range(0, 9) as $x) {
+            foreach (range(0, 9) as $y) {
+                $triggerViews[] = $this->createForm(ShootShipType::class, [
+                    'game_hash' => $game->getHash(),
+                    'coordinates' => join(",", [$x, $y])
+                ])->createView();
+            }
+        }
+
+        return $triggerViews;
+    }
+
+    /**
+     * Shoot at opponent fleet and set correct message fro the player
+     * 
+     * @param array $data
+     * @param Game $game
+     * 
+     * @return null
+     */
+    private function manageShoot(array $data, Game $game): void
```

Comment:
"manage" is too generic. Please find a better name like "shotOpponentFleet". 
Moreover `$data` doesn't tell what is it, and you only uses coordinates, so you can pass them directly